frontend/src/metabase-lib/queries/StructuredQuery.ts

```diff
@@ -321,7 +321,7 @@ class StructuredQuery extends AtomicQuery {
   }
 
   isValid() {
-    if (!this.hasData()) {
+    if (!this.table()) {
```

Comment:
1. We're losing semantics (`hasData`) with this change.
2. Inlining a deprecated function into another one, makes the other function deprecated.

Reply:
1. we can solve this by binding the expression to the const `hasData`
2. `isValid()` should be deprecated, imo.

I'll see if there's a way to migrate `isValid()`, but the most problematic part in all this is `getStructuredQueryTable` helper.

Reply:
> I'll see if there's a way to migrate `isValid()`, but the most problematic part in all this is `getStructuredQueryTable` helper.

I don't think it deserves migration. We should just remove it once it's not used anymore.

Reply:
@kamilmielnik https://github.com/metabase/metabase/pull/38522 🎉